code/game/objects/items/janitor_key.dm

```diff
@@ -22,6 +22,12 @@
 /obj/item/access_key/Initialize(mapload)
 	. = ..()
 	RegisterSignal(SSdcs, COMSIG_ON_DEPARTMENT_ACCESS, PROC_REF(department_access_given))
+	GLOB.janitor_devices += src
+
+/obj/item/access_key/Destroy()
+	UnregisterSignal(SSdcs, COMSIG_ON_DEPARTMENT_ACCESS)
```

Comment:
I missed this so I'm necroposting 

You shouldn't unregister on destroy, unless you're a /turf. Signals will clean themselves up just fine you don't need to do it manually. Leads to overly verbose and potentially confusing code